x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/derivative.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { FormattedIndexPatternColumn, ReferenceBasedIndexPatternColumn } from '../column_types';
+import { IndexPatternLayer } from '../../../types';
+import { checkForDateHistogram, dateBasedOperationToExpression, hasDateField } from './utils';
+import { OperationDefinition } from '..';
+
+const ofName = (name: string) => {
+  return i18n.translate('xpack.lens.indexPattern.derivativeOf', {
+    defaultMessage: 'Derivative of {name}',
+    values: { name },
+  });
+};
+
+export type DerivativeIndexPatternColumn = FormattedIndexPatternColumn &
+  ReferenceBasedIndexPatternColumn & {
+    operationType: 'derivative';
+  };
+
+export const derivativeOperation: OperationDefinition<
+  DerivativeIndexPatternColumn,
+  'fullReference'
+> = {
+  type: 'derivative',
+  priority: 2,
+  displayName: i18n.translate('xpack.lens.indexPattern.derivative', {
+    defaultMessage: 'Derivative',
```

Comment:
We didn't make a final decision about this, but maybe this is a good time. Can we rename this to "Differences"? It's similar to "Deltas" but I prefer "Differences"

Reply:
Good point, I like "Differences"